core/src/main/java/org/polypheny/db/adapter/Scannable.java

```diff
@@ -0,0 +1,225 @@
+import org.polypheny.db.algebra.type.GraphType;
+import org.polypheny.db.catalog.IdBuilder;
+import org.polypheny.db.catalog.catalogs.StoreCatalog;
+import org.polypheny.db.catalog.entity.allocation.AllocationCollection;
+import org.polypheny.db.catalog.entity.allocation.AllocationColumn;
+import org.polypheny.db.catalog.entity.allocation.AllocationEntity;
+import org.polypheny.db.catalog.entity.allocation.AllocationGraph;
+import org.polypheny.db.catalog.entity.allocation.AllocationTable;
+import org.polypheny.db.catalog.entity.allocation.AllocationTableWrapper;
+import org.polypheny.db.catalog.entity.logical.LogicalCollection;
+import org.polypheny.db.catalog.entity.logical.LogicalColumn;
+import org.polypheny.db.catalog.entity.logical.LogicalEntity;
+import org.polypheny.db.catalog.entity.logical.LogicalGraph;
+import org.polypheny.db.catalog.entity.logical.LogicalTable;
+import org.polypheny.db.catalog.entity.logical.LogicalTableWrapper;
+import org.polypheny.db.catalog.entity.physical.PhysicalEntity;
+import org.polypheny.db.catalog.entity.physical.PhysicalTable;
+import org.polypheny.db.catalog.exceptions.GenericRuntimeException;
+import org.polypheny.db.catalog.logistic.Collation;
+import org.polypheny.db.catalog.logistic.PlacementType;
+import org.polypheny.db.prepare.Context;
+import org.polypheny.db.schema.trait.ModelTrait;
+import org.polypheny.db.tools.AlgBuilder;
+import org.polypheny.db.type.PolyType;
+import org.polypheny.db.util.Triple;
+
+public interface Scannable {
+
+    static PhysicalTable createSubstitutionTable( Scannable scannable, Context context, LogicalEntity logical, AllocationEntity allocation, String name, List<Triple<String, Integer, PolyType>> nameLength ) {
+        IdBuilder builder = IdBuilder.getInstance();
+        LogicalTable table = new LogicalTable( builder.getNewLogicalId(), name + logical.id, logical.namespaceId, logical.entityType, null, logical.modifiable );
+        List<LogicalColumn> columns = new ArrayList<>();
+
+        int i = 0;
+        for ( Triple<String, Integer, PolyType> col : nameLength ) {
+            LogicalColumn column = new LogicalColumn( builder.getNewFieldId(), col.getLeft(), table.id, table.namespaceId, i, col.getRight(), null, col.getMiddle(), null, null, null, false, Collation.getDefaultCollation(), null );
+            columns.add( column );
+            i++;
+        }
+        AllocationTable allocSubTable = new AllocationTable( builder.getNewAllocId(), allocation.placementId, allocation.partitionId, table.id, table.namespaceId, allocation.adapterId );
+
+        List<AllocationColumn> allocColumns = new ArrayList<>();
+
+        for ( LogicalColumn column : columns ) {
+            AllocationColumn alloc = new AllocationColumn( logical.namespaceId, allocSubTable.placementId, allocSubTable.logicalId, column.id, PlacementType.AUTOMATIC, column.position, allocation.adapterId );
+            allocColumns.add( alloc );
+        }
+        // we use first as pk
```

Comment:
This is how we also handled it before. However, isn't there a more elegant solution?

Reply:
changed to allow for composite PKs, still requires them to be positioned in front, to not pollute this method.